tests/GitWrapper/GitRepositoryTest.php

```diff
@@ -176,6 +176,29 @@ public function testHasChangesFindsChanges()
         $this->assertTrue($this->gitRemoteRepository->hasChanges());
     }
 
+    public function testTagReturnsTags()
+    {
+        $localRepository = $this->initializeLocalRepository();
+        $repositoryDirectory = rtrim($localRepository->getDirectory(), '/\\') . '/';
+        $tagStrings = [
+            '0.3.1-test',
+            'v1.4.1-rc_34123',
+            '1.0.1',
+        ];
+        foreach ($tagStrings as $tagString) {
+            $this->gitWrapper->execute('tag', [['a' => $tagString], ['m' => 'Tag Version']], [], $repositoryDirectory);
+        }
+
+        $this->assertSame(
+            [
+                'v1.4.1-rc_34123',
+                '1.0.1',
+                '0.3.1-test',
+            ],
+            array_reverse($localRepository->tag(['l']))
```

Comment:
`git tag -l` reorder the tags itself. 
the reverse was simply to be compensate the missing option in git under 2.0. 
yes, It not really necessary.